doc/rados/operations/placement-groups.rst

```diff
@@ -390,172 +407,162 @@ group will store objects on two OSDs, as shown below.
+lower the risk of permanently losing a PG due to cascading failures. As far as
+data durability is concerned, in a cluster with fewer than 50 OSDs, it doesn't
+much matter whether there are 512 or 4096 PGs.
+
+.. note::  It can take a long time for an OSD that has been recently added to
+   the cluster to be populated with the PGs that were assigned to it.  However,
+   no object degradation or impact on data durability will result from the
+   slowness of this process.
 
 .. _object distribution:
 
 Object distribution within a pool
 ---------------------------------
 
-Ideally objects are evenly distributed in each placement group. Since
-CRUSH computes the placement group for each object, but does not
-actually know how much data is stored in each OSD within this
-placement group, the ratio between the number of placement groups and
-the number of OSDs may influence the distribution of the data
-significantly.
-
-For instance, if there was a single placement group for ten OSDs in a
-three replica pool, only three OSD would be used because CRUSH would
-have no other choice. When more placement groups are available,
-objects are more likely to be evenly spread among them. CRUSH also
-makes every effort to evenly spread OSDs among all existing Placement
-Groups.
-
-As long as there are one or two orders of magnitude more Placement
-Groups than OSDs, the distribution should be even. For instance, 256
-placement groups for 3 OSDs, 512 or 1024 placement groups for 10 OSDs
-etc.
-
-Uneven data distribution can be caused by factors other than the ratio
-between OSDs and placement groups. Since CRUSH does not take into
-account the size of the objects, a few very large objects may create
-an imbalance. Let say one million 4K objects totaling 4GB are evenly
-spread among 1024 placement groups on 10 OSDs. They will use 4GB / 10
-= 400MB on each OSD. If one 400MB object is added to the pool, the
-three OSDs supporting the placement group in which the object has been
-placed will be filled with 400MB + 400MB = 800MB while the seven
-others will remain occupied with only 400MB.
+Under ideal conditions, objects are evenly distributed in each PG. Because
+CRUSH computes the PG for each object but does not know how much data is stored
+in each OSD associated with the PG, the ratio between the number of PGs and the
+number of OSDs can have a significant influence on data distribution.
+
+For example, suppose that there is only a single PG for ten OSDs in a
+three-replica pool. In that case, only three OSDs would be used because CRUSH
+would have no other option. However, if more PGs are available, objects are
+more likely to be evenly distributed across OSDs.  CRUSH makes every effort to
+distribute OSDs evenly across all existing PGs.
+
+As long as there are one or two orders of magnitude more PGs than OSDs, the
+distribution is likely to be even. For example: 256 PGs for 3 OSDs, 512 PGs for
+10 OSDs, or 1024 PGs for 10 OSDs.
```

Comment:
Maybe add a note about how `pg_num` not being a power of 2 can work against uniform distribution.  Don't go into why, if someone really cares they can read chapter 8 of my book, so long as they pre-dose with their favorite headache medication.

Reply:
This is covered later on this same page (and will be edited in the "3 of x" PR in this series).